cli/tests/unit/promise_hooks_test.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+import { assertEquals } from "./test_util.ts";
+
+const twoAwaitResolveThenRejectWithCatch = [
+  "init",
+  "init",
+  "resolve",
+  "init",
+  "init",
+  "init",
+  "init",
+  "init",
+  "init",
+  "init",
+  "init",
+  "before",
+  "init",
+  "resolve",
+  "init",
+  "resolve",
+  "after",
+  "before",
+  "init",
+  "resolve",
+  "init",
+  "resolve",
+  "resolve",
+  "after",
+  "before",
+  "resolve",
+  "after",
+  "before",
+];
```

Comment:
These don't seem to match up with #8209. Can you explain or comment the code so it's more clear how this matches up with the code generating the events?